bqskit/passes/partitioning/quick.py

```diff
@@ -191,6 +212,32 @@ def process_pending_bins() -> None:
                 pending_bins.append(BarrierBin(point, location, circuit))
                 continue
 
+            # Measurement close all overlapping bins
+            if isinstance(op.gate, MeasurementPlaceholder):
+                for bin in overlapping_bins:
+                    if close_bin_qudits(bin, location, cycle):
+                        num_closed += 1
+                    else:
+                        extended = [q for q in location if q not in bin.qudits]
+                        bin.blocked_qudits.update(extended)
+
+                # Track the measurement to restore it in partitioned circuit
+                pending_bins.append(MeasurementBin(point, location, circuit))
+                continue
+
+            # Reset close all overlapping bins
+            if isinstance(op.gate, Reset):
+                for bin in overlapping_bins:
+                    if close_bin_qudits(bin, location, cycle):
+                        num_closed += 1
+                    else:
+                        extended = [q for q in location if q not in bin.qudits]
+                        bin.blocked_qudits.update(extended)
```

Comment:
The [DRY software](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) engineering principle applies here. I am not always the biggest fan of it, but definitely extract shared functionality, especially if exactly equal code, to a method or accumulate the if blocks into one. In this case, when the code is unnecessarily elongated, it makes it more difficult and time-consuming for future readers to understand the purpose of each code section.